cpp/arcticdb/processing/clause.cpp

```diff
@@ -367,6 +380,11 @@ Composite<EntityIds> AggregationClause::process(Composite<EntityIds>&& entity_id
     // Work out the common type between the processing units for the columns being aggregated
     procs.broadcast([&aggregators_data, &aggregators=aggregators_](auto& proc) {
         for (auto agg_data: folly::enumerate(aggregators_data)) {
+            // Check that segments row ranges are the same
+            internal::check<ErrorCode::E_ASSERTION_FAILURE>(
+                std::all_of(proc.row_ranges_->begin(), proc.row_ranges_->end(), [&] (const auto& row_range) {return row_range->start() == proc.row_ranges_->at(0)->start();}),
```

Comment:
IIUC, checking on `start` only would be enough.

Reply:
For now yes, if the start is the same the end will be too